product/util.py

```diff
@@ -154,15 +157,52 @@ def merge_products(left_product, right_product):
     return merged
 
 
-def diff_products(left_product, right_product, ignore_component_names=()):
+@dataclasses.dataclass
+class ComponentDiff:
+    cidentities_only_left: set = dataclasses.field(default_factory=set)
+    cidentities_only_right: set = dataclasses.field(default_factory=set)
+    cpairs_version_changed: list = dataclasses.field(default_factory=list)
+    # only set when new component is added/removed
+    names_only_left: set = dataclasses.field(default_factory=set)
+    names_only_right: set = dataclasses.field(default_factory=set)
+    # only set on update
+    names_version_changed: set = dataclasses.field(default_factory=set)
+
+
+# for v2 =  pass cd v2 and resolve components
+# make use of caching
+def diff_products(
+    left_product: cm.ComponentDescriptor,
```

Comment:
I know, the parameter names are originally mine (back then, the component-descriptors were called (by me) "product" for some reason)

how about (since this is an incompatible change anyhow) renaming this to `left_component` / `right_component`?

also, we _might_ be a bit nice here, and e.g. accept either a `Component` or a `ComponentDescriptor`, wdyt?

Reply:
done